docs/mcap/README.md

```diff
@@ -0,0 +1,371 @@
+| 8+N | data | uint64 length-prefixed bytes |  Size of the attachment. |
+| 4 | crc | uint32 | CRC of preceding fields in the record. Optionally zero. |
+
+#### Attachment Index (op=0x09)
+The attachment index is an index to named attachments within the file. One
+record is recorded per attachment in the file.
+
+| Bytes | Name | Type | Description
+| --- | --- | --- | --- |
+| 8 | record_time | Timestamp | Timestamp at which the attachment was recorded.
+| 8 | attachment_size | uint64 | Size of the attachment. |
+| 4 + N | name | String | Name of the attachment. |
+| 4 + N | content_type | String | MIME type of the attachment. |
+| 8 | offset | uint64 | Byte offset to the attachment, from the start of the file. |
+
+#### Statistics (op=0x0A)
+The statistics record contains statistics about the recorded data. It is the
+last record in the file before the footer.
+
+| Bytes | Name | Type | Description |
+| --- | --- | --- | --- |
+| 8 | message_count | uint64 | Number of messages in the file across all topics. |
+| 4 | channel_count | uint32 | Number of channels in the file across all topics. |
+| 4 | attachment_count | uint32 | Number of attachments in the file. | 
+| 4 | chunk_count | uint32 | Number of chunks in the file. |
+| N | channel_stats | KeyValues<uint16, uint64> | Array of channel IDs and total message counts for the channel. |
+
+### Feature Explanations
+The format is intended to support efficient, indexed reading of messages and
+generation of summary data in both local and remote contexts. "Seeking" should
+be imagined to incur either a disk seek or an HTTP range request to an object
+store -- the latter being significantly more costly.
+
+#### Scanning for records on specific topics within an interval
+The index is designed to support fast local and remote seek/filter operations
+with minimal seeking or range request overhead. The operation of the index for
+message reading is as follows:
+1. Client queries for all messages on topics /a, /b, /c between t0 and t1
+2. Reader reads the fixed-length footer off the end of the file
+3. Reader parses the index_offset from the footer, and starts reading from that
+   offset to the end of the file. During this read it will encounter the
+   following in order:
+   * A run of channel info records, one per channel in the file
+   * A run of Message Group Index records, one per chunk in the file
+   * The attachment index records
+   * The statistics record
+
+The reader in this case will stop after the chunk index records.
+
+4. Using the channel info records at the start of the read, the reader converts
+topic names to channel IDs.
+5. Using the chunk index records, the reader locates the chunks that must be read,
+based on the requested start times, channel IDs, and end times. These chunks
+will be a contiguous run.
+6. Readers may access the message data in at least two ways,
+   * “full scan”: Seek from the chunk index the start of the chunk using
```

Comment:
Not sure I understand this sentence, seems like there is at least a typo: "from the chunk index the start of the chunk"